starlite/handlers/http_handlers.py

```diff
@@ -192,8 +191,8 @@ def _create_data_handler(
     ]
     cookie_headers = [cookie.to_encoded_header() for cookie in cookies if not cookie.documentation_only]
     raw_headers = [*normalized_headers, *cookie_headers]
-    is_dto_annotation = is_class_and_subclass(return_annotation, DTO)
-    is_dto_iterable_annotation = annotation_is_iterable_of_type(return_annotation, DTO)
+    is_dto_annotation = is_class_and_subclass(return_annotation, dto.Factory)
+    is_dto_iterable_annotation = annotation_is_iterable_of_type(return_annotation, dto.Factory)
```

Comment:
these are failing in the examples when the return annotation is a forward ref due to `from __future__ import annotations` - Signature.from_callable() isn't resolving the runtime type, so these are always false under those conditions.